### PR TITLE
release: Prepare for 0.4.0

### PR DESCRIPTION
**Pull request by real-or-random**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,18 +5,26 @@ All notable changes to this project will be documented in this file.
...
-## [Unreleased]
+## [0.4.0] - 2023-09-04
 
 #### Added
  - New module `ellswift` implements ElligatorSwift encoding for public keys and x-only Diffie-Hellman key exchange for them.
    ElligatorSwift permits representing secp256k1 public keys as 64-byte arrays which cannot be distinguished from uniformly random. See:
    - Header file `include/secp256k1_ellswift.h` which defines the new API.
    - Document `doc/ellswift.md` which explains the mathematical background of the scheme.
    - The [paper](https://eprint.iacr.org/2022/759) on which the scheme is based.
+ - We now test the library with unreleased development snapshots of GCC and Clang. This gives us an early chance to catch miscompilations and constant-time issues introduced by the compiler (such as these that led to the previous two releases).
```
Comment (jonasnick) [inline CHANGELOG.md:16]: these -> those?

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+#### Fixed
+ - Certain internal library symbols are no longer exported on Windows DLL builds.
```
Comment (hebasto) [inline CHANGELOG.md:19]: Maybe drop the word "Certain"?

I'm not a native English speaker, but the sentence in its current variant makes me think that there are still internal symbols that are exported.

Comment (real-or-random) [reply CHANGELOG.md]: rephrased

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,22 +5,25 @@ All notable changes to this project will be documented in this file.
...
+## [0.4.0] - 2023-09-04
 
 #### Added
  - New module `ellswift` implements ElligatorSwift encoding for public keys and x-only Diffie-Hellman key exchange for them.
    ElligatorSwift permits representing secp256k1 public keys as 64-byte arrays which cannot be distinguished from uniformly random. See:
    - Header file `include/secp256k1_ellswift.h` which defines the new API.
    - Document `doc/ellswift.md` which explains the mathematical background of the scheme.
    - The [paper](https://eprint.iacr.org/2022/759) on which the scheme is based.
- - We now test the library with unreleased development snapshots of GCC and Clang. This gives us an early chance to catch miscompilations and constant-time issues introduced by the compiler (such as these that led to the previous two releases).
+ - We now test the library with unreleased development snapshots of GCC and Clang. This gives us an early chance to catch miscompilations and constant-time issues introduced by the compiler (such as those that led to the previous two releases).
```
Comment (hebasto) [inline CHANGELOG.md:16]: Sorry for nit-picking, but this change belongs to the previous commit, no?

Comment (real-or-random) [reply CHANGELOG.md]: fixed

**Review by hebasto — APPROVED** [review]
ACK 50e7da635c0212effa6d20771f6cfc1a7a0b092f

**Review by jonasnick — APPROVED** [review]
ACK 50e7da635c0212effa6d20771f6cfc1a7a0b092f

**Comment by hebasto** [comment]
re-ACK 16339804c906a99a0b9bbe192d3de9fde021ff19.

**Review by jonasnick — APPROVED** [review]
ACK 16339804c906a99a0b9bbe192d3de9fde021ff19

**Pull request merged**